ui/raidboss/emulator/data/CombatantState.ts

```diff
@@ -0,0 +1,88 @@
+export interface ICombatantState {
+  posX?: number;
+  posY?: number;
+  posZ?: number;
+  heading?: number;
+  targetable?: boolean;
+  HP?: number;
+  maxHP?: number;
+  MP?: number;
+  maxMP?: number;
+}
+
+// Member names taken from OverlayPlugin's MiniParse.cs
+// Types taken from FFXIV parser plugin
+export interface IPluginState {
+  CurrentWorldID?: number;
+  WorldID?: number;
+  WorldName?: string;
+  BNpcID?: number;
+  BNpcNameID?: number;
+  PartyType?: number;
+  ID?: number;
+  OwnerID?: number;
+  type?: number;
+  Job?: number;
+  Level?: number;
+  Name?: string;
+  CurrentHP: number;
+  MaxHP: number;
+  CurrentMP: number;
+  MaxMP: number;
+  PosX: number;
+  PosY: number;
+  PosZ: number;
+  Heading: number;
+}
+
+export default class CombatantState implements ICombatantState {
+  posX: number;
+  posY: number;
+  posZ: number;
+  heading: number;
+  targetable: boolean;
+  HP: number;
+  maxHP: number;
+  MP: number;
+  maxMP: number;
+
+  constructor(posX: number, posY: number, posZ: number, heading: number,
+      targetable: boolean,
+      HP: number, maxHP: number, MP: number, maxMP: number) {
+    this.posX = posX;
+    this.posY = posY;
+    this.posZ = posZ;
+    this.heading = heading;
+    this.targetable = targetable;
+    this.HP = HP;
+    this.maxHP = maxHP;
+    this.MP = MP;
+    this.maxMP = maxMP;
+  }
+
+  partialClone(props: ICombatantState): CombatantState {
```

Comment:
```suggestion
  partialClone(props: Partial<CombatantState>): CombatantState {
```

`ICombatantState` looks like it's mostly repeating `CombatantState`.  What about using `Partial` and only having one type?